### PR TITLE
Major overhaul of `vault operator generate-root` CLI help

### LINKED ISSUE #15252: Issues with `vault operator generate-root` CLI help message
This part of the CLI help message is incorrect: https://github.com/hashicorp/vault/blob/967b2987e5a17a82f9b46b6d0b4f93b42fd550cd/command/operator_generate_root.go#L80-L82

The `-otp` flag is not needed or used, when providing an unseal key shard. It is only for use in conjunction with `-init` or `-decode`.

This earlier part of the CLI help message could be made more useful - it documents that you must use a multi-step process to begin an OTP-based root token generation:
* `vault operator generate-root -generate-otp`
* `vault operator generate-root -init -otp="..."`

In fact you can use just:
`vault operator generate-root -init`
as shorthand for both steps in one command.

It would be nice to list this workflow option first, as it is the most simple and convenient for someone reading the help message with an immediate need to generate a root token. The other workflows are more relevant to someone implementing a more complex and pre-planned key ceremony.

### PR DESCRIPTION
**Pull request by maxb**
Resolves #15252

A major overhaul of the `vault operator generate-root` CLI help to
surface the fact that it is actually six separate commands in one,
rather than requiring users to independently deduce this mental model
themselves.

In the process of doing so, also standardize some terminology:

* Fix places which used the phrase "operational token" instead of
  "operation token" to be consistent with the prevailing terminology.

* Fix places which used the phrase "recovery operation token" instead of
  "recovery token" to be consistent with the prevailing terminology.

This PR currently focusses on the CLI help, but following review and
feedback, I assume I'll need to replicate many of the same changes in
website/content/docs/commands/operator/generate-root.mdx as well.

### CODE REVIEW

**Comment by maxb** [comment]
Please consider labelling `pr/no-changelog`, as this changes only help strings, comments, log message strings and docs.

**Comment by hsimon-hashicorp** [comment]
This also looks like it needs a re-base, @maxb, and I'll put it in front of our engineers in the meantime. :)

**Comment by VioletHynes** [comment]
This looks great! Thanks, Max. I'm going to adjust a little of the wording before I merge this, but I'll try and get this merged shortly.

Appreciate the PR! This should make things a lot clearer.

File: command/operator_generate_root.go [hunk 1]
```diff
@@ -49,40 +49,99 @@ type OperatorGenerateRootCommand struct {
...
 
-    - A base64-encoded one-time-password (OTP) provided via the "-otp" flag.
-      Use the "-generate-otp" flag to generate a usable value. The resulting
-      token is XORed with this value when it is returned. Use the "-decode"
-      flag to output the final value.
+  This command is unusual, as it is effectively six separate subcommands, but
+  instead of being implemented as actual subcommands, there are six different
+  forms of the command, selected via the options -init, -decode,
+  -generate-otp, -status, -cancel, or the absence of any of the previous five
+  options (which selects the "provide a key share" form).
```
Comment (VioletHynes) [inline command/operator_generate_root.go:70]:
```suggestion
  This command is unusual, as it is effectively six separate subcommands,
  selected via the options -init, -decode, -generate-otp, -status, -cancel, 
  or the absence of any of the previous five options (which selects the 
  "provide a key share" form).
```

**Review by VioletHynes — APPROVED** [review]

**Pull request merged**